### ISSUE TITLE
Create proof-of-concept web widget

### ISSUE THREAD

**Issue by hannobraun** [hannobraun/fornjot#816]
One of Fornjot's goals is support for embedding Fornjot models into websites, without any need for a custom backend. For that, we need some kind of widget that can be embedded in a website to display a Fornjot model there. Presumably, this widget would include a WebAssembly module that contains most of the Fornjot components.

The object of this issue is to create such a widget as a proof of concept and basis for further iteration. It doesn't need to be perfect, or even good. It just needs to demonstrate that embedding Fornjot models into websites is something that can work.

I assume that this widget would be a thin Rust crate that uses other Fornjot components as required to do the actual work, similar to `fj-app`. Initially, the ability to load models is not required. Just ship the thing with a simple default model, to demonstrate that it can work. Eventually, we will most likely use WebAssembly to load models (see #71).

This issue is currently blocked on #815.

**Comment by hannobraun** [comment]
#815 has been addressed. This issue is no longer blocked.

**Comment by OliverEvans96** [comment]
I just want to express my excitement for this pursuit. It would be really cool to view models in the browser.

Admittedly, I haven't actually looked at the fj-viewer, so I'm not sure if it includes the ability to modify parameters. But a fascinating next step (after dynamically loading models is implemented) would be to expose a basic JS API so that parameters can be modified via HTML sliders, etc.

**Comment by hannobraun** [comment]
Thank you for the comment, @OliverEvans96!

`fj-viewer` doesn't currently support modifying parameters, but that's definitely planned (https://github.com/hannobraun/Fornjot/issues/35). And yes, you're absolutely correct, that capability should also be present in the browser!

**Comment by hannobraun** [comment]
Just a note, so the information is available once implementation on this starts: [wgpu](https://crates.io/crates/wgpu) has some `async` functions which need to be handled. That is in addition to any other async Rust code we might add in the meantime.

We're probably going to need different executors in the browser and elsewhere, and the combination of [Pollster](https://crates.io/crates/pollster) and [`wasm-bindgen-futures`](https://crates.io/crates/wasm-bindgen-futures) [seems to work well](https://twitter.com/tomhemmes/status/1611384107669540866).

**Comment by hannobraun** [comment]
Closing as out of scope. See [A New Direction](https://www.fornjot.app/blog/a-new-direction/) for context.
